File: fab-core/src/main/java/eu/stratosphere/fab/core/beans/system/ExperimentRunner.java
package eu.stratosphere.fab.core.beans.system;

import eu.stratosphere.fab.core.beans.ExecutionContext;

import java.util.Set;

abstract public class ExperimentRunner extends System {

    public ExperimentRunner(Lifespan lifespan, Set<System> dependencies) {
        super(lifespan, dependencies);
    }

    abstract public void run(ExecutionContext context);
}

File: fab-core/src/main/scala/eu/stratosphere/fab/core/beans/system/ExperimentRunner.scala
package eu.stratosphere.fab.core.beans.system

import eu.stratosphere.fab.core.beans.system.Lifespan.Lifespan
import eu.stratosphere.fab.core.ExecutionContext
import java.io.File


/**
 * Created by felix on 02.06.14.
 */
abstract class ExperimentRunner(lifespan: Lifespan, dependencies: Set[System])
  extends System(lifespan, dependencies) {

  def run(job: String)

}
